### DISCUSSION TITLE
Upgraded to the latest Pocketbase and I get the error that my db is malformed

### DISCUSSION THREAD

**Discussion by MuslemRahimi** [pocketbase/pocketbase#6081] [Q&A]
Hi,

as the title suggested I updated my pocketbase and js-sdk to the latest build.

I suddenly get random errors and crashes 

```
database disk image is malformed (11); failed query: SELECT `_authOrigins`.* FROM `_authOrigins` WHERE `collectionRef`={:p0} AND `recordRef`={:p1} ORDER BY `created` DESC
```

```
Something went wrong while processing your request.
disk I/O error (522); failed query: SELECT {{_logs}}.* FROM `_logs` ORDER BY [[_logs]].[[_rowid_]] DESC LIMIT 50
```

I need to restart pocketbase to make it work again. Any idea what I can do to fix this issue?

**Comment by ganigeorgiev** [comment]
Without clear steps to reproduce it I'm not sure how to help as the "database disk image is malformed" could be returned for a lot of reasons.

Make sure to follow the upgrade instructions exactly as per the [CHANGELOG](https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md) and more specifically the "Upgrade overview" section in:
- https://pocketbase.io/v023upgrade/go/
- https://pocketbase.io/v023upgrade/jsvm/

I also remember you from https://github.com/pocketbase/pocketbase/discussions/3867 and https://github.com/pocketbase/pocketbase/discussions/4377, so if you are still using Marmot or other similar solution make sure that it is not related to it.

If you have already followed the steps in the changelog exactly as it is and it is still failing, then try to create a **minimal** reproducible repo or steps how to trigger the above error.

**Reply by MuslemRahimi** [reply]
Thanks for the quick reply. I will try to create a minimal reproducible rep. I deleted marmot completely from pocketbase.

**Reply by ganigeorgiev** [reply]
I've checked the journalctl, stderr and db logs of my production instance in app.presentator.io and the live PocketBase demo and in both cases there is no indication for crashes or db errors.

Your issue is most likely something specific to your environment and deployment setup but again, the error is too generic to try to pinpoint the exact cause of it. It could be lack of disk space, limited file descriptors, permission/antivirus software issue, misuse of 3rd party app that operates directly with the SQLite file, network mounted drive misconfiguration, etc.

If you haven't already, you can check ["How To Corrupt An SQLite Database File"](https://sqlite.org/howtocorrupt.html).

You can also try to connect directly to the `pb_data/data.db` with the sqlite3 cli tool and run `PRAGMA integrity_check;` to see whether it will fail.

**Reply by MuslemRahimi** [reply]
I get 
```
sqlite3 data.db "PRAGMA integrity_check;"
*** in database main ***
Page 1160: btreeInitPage() returns error code 11
Page 874: btreeInitPage() returns error code 11
Page 70: btreeInitPage() returns error code 11
Error: stepping, database disk image is malformed (11)
```

**Reply by ganigeorgiev** [reply]
This confirm that db file is corrupted but I don't know what has caused it.

Again, I doubt that it is related to the upgrade because as mentioned earlier I don't have this issue on 2 production environments.

Your posts history suggest that there is something strange going on with your environment but without detailed steps about how you are running PocketBase, how do you backup your database, how did you upgrade, are there any external apps that connects directly to the db file (e.g. Litestream or another application), etc., I don't know how to help further.

To try to fix the above error:
1. Shutdown PocketBase and any other 3rd party app connecting to the database.
2. Via the `sqlite3` tool try to create a backup using the `.backup data2.db` command or `VACUUM INTO data2.db`.
3. Rename `data2.db` to `data.db`.
4. Delete the `auxiliary.db` file (this is the logs db and it will be recreated on startup)
5. Start the application.

**Reply by ganigeorgiev** [reply]
There is also `.recover` command. For more details please check https://www.sqlite.org/recovery.html.

**Comment by miguelgargallo** [comment]
It is realted to the upgrade, i just changed my version on docker, redeploy, and upload my backup and says this

**Reply by miguelgargallo** [reply]
So I go back to `0.22.14` and solve it

**Reply by ganigeorgiev** [reply]
Without clear steps to reproduce and details about your environment I cannot assist you and as mentioned above I've tested it on 2 different production environments (_arm64 and Intel_x86_) and it is working correctly for me.

It is also not clear what do you mean by "upload my backup" (especially if you are trying to upload a backup from an older version). Please follow the instructions exactly as it is in the linked upgrade guides above.

**Reply by ganigeorgiev** [reply]
If you are not planning to upgrade to v0.23, I recommend at least to update to the latest v0.22.29 version (you can find the <=v0.22.x CHANGELOGs in https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG_16_22.md)
